src/middlewared/middlewared/plugins/cloud_sync.py

```diff
@@ -465,6 +468,38 @@ class Config:
         if verrors:
             raise verrors
 
+    @private
+    def migrate_sftp_key_file(self):
```

Comment:
I am not sure I like the idea of running that every time system is ready or pool is imported.

It sounds like we need a one-time scripts run for upgrades?